### PR TITLE
Update api_changes.md to include section about regenerating json (un)marshaling code

### PR DESCRIPTION
**Pull request by wojtek-t**
Requested in https://github.com/kubernetes/kubernetes/pull/15072

@pmorie @lavalamp @krousey

### CODE REVIEW

**Comment by pmorie** [comment]
LGTM, thanks a lot!

File: docs/devel/api_changes.md [hunk 1]
```diff
@@ -446,12 +446,28 @@ To regenerate them:
...
+
+The auto-generated code resides with each versioned API:
+   - `pkg/api/<version>/types.generated.go`
+   - `pkg/apis/extensions/<version>/types.generated.go`
+
+To regenerate them:
+   - run
+
+```sh
+hach/update-codecgen.sh
```
Comment (thockin) [inline docs/devel/api_changes.md]: s/hach/hack

**Comment by thockin** [comment]
You can self-lgtm when typo is fixed

File: docs/devel/api_changes.md [hunk 2]
```diff
@@ -293,13 +293,13 @@ the release notes for the next release by labeling the PR with the "release-note
 If you found that your change accidentally broke clients, it should be reverted.
 
 In short, the expected API evolution is as follows:
-* `experimental/v1alpha1` ->
+* `extensions/v1alpha1` ->
```
Comment (lavalamp) [inline docs/devel/api_changes.md]: I would like to begin starting new features in their own group to begin with.

File: docs/devel/api_changes.md (hunk 1 above)
Comment (wojtek-t) [inline docs/devel/api_changes.md]: Done.

**Pull request merged**